yaksh/views.py

```diff
@@ -1264,49 +1273,80 @@ def user_data(request, user_id, questionpaper_id=None):
                                  context_instance=RequestContext(request))
 
 
+def _expand_questions(questions, field_list):
+    i = field_list.index('questions')
+    field_list.remove('questions')
+    for question in questions:
+        field_list.insert(i, '{0}-{1}'.format(question.summary, question.points))
+    return field_list 
+
+
 @login_required
 @email_verified
-def download_csv(request, questionpaper_id):
-    user = request.user
-    if not is_moderator(user):
+def download_quiz_csv(request, course_id, quiz_id):
+    current_user = request.user
+    if not is_moderator(current_user):
         raise Http404('You are not allowed to view this page!')
-    quiz = Quiz.objects.get(questionpaper=questionpaper_id)
+    course = get_object_or_404(Course, id=course_id)
+    quiz = get_object_or_404(Quiz, id=quiz_id)
+    if not course.is_creator(current_user) and not course.is_teacher(current_user):
+        raise Http404('The quiz does not belong to your course')
+    users = course.get_enrolled().order_by('first_name')
+    if not users:
+        return monitor(request, quiz_id)
+    csv_fields = []
+    attempt_number = None
+    question_paper = quiz.questionpaper_set.last()
+    last_attempt_number =AnswerPaper.objects.get_attempt_numbers(question_paper.id).last()
+    if request.method == 'POST':
+        csv_fields = request.POST.getlist('csv_fields')
+        attempt_number = request.POST.get('attempt_number', last_attempt_number)
+    if not csv_fields:
+        csv_fields = ['name', 'username', 'roll_number', 'institute',
+                'department', 'questions', 'total', 'out_of', 'percentage', 'status']
```

Comment:
Just a suggestion instead of `total` can we write it as `marks_obtained`?

Reply:
Changed